robocorp-code/vscode-client/src/extension.ts

```diff
@@ -324,6 +335,85 @@ async function cloudLogoutAndRefresh() {
     refreshCloudTreeView();
 }
 
+async function convertProject() {
+    const DEFAULT_ERROR_MSG = `
+            Could not convert project.
+            Please check the output logs for more details.
+            `;
+
+    const DEFAULT_ERROR_STATUS = "Error while converting project.";
+
+    try {
+        const converterLocation = await ensureConvertBundle(true);
+        if (!converterLocation) {
+            throw new Error("Converter bundle was not found. Possible download issues.");
+        }
+        const converterBundle = require(converterLocation);
+
+        // let the user decide what type of project will be converted
+        const vendorMap = {
+            "Blue Prism": "blueprism",
+            "UiPath": "uipath",
+            "Automation Anywhere 360": "a360",
+        };
+        const items = Object.keys(vendorMap);
+        let selectedVendor = await window.showQuickPick(items, {
+            "placeHolder": "Please select the format of current project",
+            "canPickMany": false,
+            "ignoreFocusOut": true,
+        });
+        if (!selectedVendor) {
+            // exit conversion if there is no item selected
+            return;
+        }
+
+        // let the user decide where the conversion result will be saved
+        let wsFolders: ReadonlyArray<WorkspaceFolder> = workspace.workspaceFolders;
+        let ws: WorkspaceFolder;
+        if (wsFolders.length == 1) {
+            ws = wsFolders[0];
+        } else {
+            ws = await window.showWorkspaceFolderPick({
+                "placeHolder": "Please select the workspace folder for the converter output",
+                "ignoreFocusOut": true,
+            });
+        }
+        if (!ws) {
+            // Operation cancelled.
+            return;
+        }
+        const destination = Uri.joinPath(ws.uri, "converted");
+        const activeTextEditor = window.activeTextEditor;
+        const conversionResult: ConversionResult = await converterBundle.convert(
+            vendorMap[selectedVendor],
+            activeTextEditor.document.getText()
+        );
+        if (!converterBundle.isSuccessful(conversionResult) && conversionResult.type === "Failure") {
```

Comment:
Should this be an `||` instead of `&&`?

Reply:
The operators are mostly equivalent, but because the bundle & its contents are hidden, Typescript will complain when dealing with values that are in either the ConversionSuccess | ConversionError.

Reply:
I'm not sure I follow... if they're mostly the same, I guess you can just do: `!converterBundle.isSuccessful(conversionResult)` and not the `&& conversionResult.type === "Failure"` part, right?